test/functional/test_framework/test_node.py

```diff
@@ -96,6 +96,15 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-debugexclude=leveldb",
             "-uacomment=testnode%d" % i,
         ]
+        if use_valgrind:
+            default_suppressions_file = os.path.join(
+                os.path.dirname(os.path.realpath(__file__)),
+                "..", "..", "..", "contrib", "valgrind.supp")
+            suppressions_file = os.getenv("VALGRIND_SUPPRESSIONS_FILE",
+                                          default_suppressions_file)
+            self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
+                         "--gen-suppressions=all", "--exit-on-first-error=yes",
```

Comment:
`--exist-on-first-error` might slow down the development cycle if there are multiple valgrind problems in a single change.

Reply:
There is no easy way to achieve what we are trying to do here without using `--exist-on-first-error`, and hopefully it won't be very common with more than one memory problem per `bitcoind` execution. Additionally, continuing execution after a memory error is not meaningful: the diagnostics emitted after the first error are likely spurious anyways :)

So in summary -- we'll have to live with `--exit-on-first-error=yes` :)